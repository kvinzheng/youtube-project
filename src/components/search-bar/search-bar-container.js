import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loadAllData } from '../../redux/actions/index';
import SearchBar from './search-bar';

export const mapDispatchToProps = dispatch => bindActionCreators({
  loadAllData,
}, dispatch);

export default connect(null, mapDispatchToProps)(SearchBar);
