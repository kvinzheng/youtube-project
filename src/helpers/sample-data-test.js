export const video = {
  "kind": "youtube#searchResult",
  "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/Cdphs-sy_uELwbqUW-uWHjQcnqo\"",
  "id": {
    "kind": "youtube#video",
    "videoId": "hY7m5jjJ9mM"
  },
  "snippet": {
    "publishedAt": "2017-05-31T09:30:02.000Z",
    "channelId": "UC9obdDRxQkmn_4YpcBMTYLw",
    "title": "CATS will make you LAUGH YOUR HEAD OFF - Funny CAT compilation",
    "description": "HELP HOMELESS ANIMALS AND WIN A PRIZE BY CHOOSING AN AWESOME ITEM FROM OUR STORE: https://fanjoy.co/collections/tiger-productions ...",
    "thumbnails": {
      "default": {
        "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/default.jpg",
        "width": 120,
        "height": 90
      },
      "medium": {
        "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/mqdefault.jpg",
        "width": 320,
        "height": 180
      },
      "high": {
        "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/hqdefault.jpg",
        "width": 480,
        "height": 360
      }
    },
    "channelTitle": "Tiger FunnyWorks",
    "liveBroadcastContent": "none"
  }
};

export const videos = {
  "kind": "youtube#searchListResponse",
  "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/qdDGutV8AE1VGqwVGO3M3L31giU\"",
  "nextPageToken": "CAUQAA",
  "regionCode": "US",
  "pageInfo": {
    "totalResults": 1000000,
    "resultsPerPage": 5
  },
  "items": [
    {
      "kind": "youtube#searchResult",
      "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/Cdphs-sy_uELwbqUW-uWHjQcnqo\"",
      "id": {
        "kind": "youtube#video",
        "videoId": "hY7m5jjJ9mM"
      },
      "snippet": {
        "publishedAt": "2017-05-31T09:30:02.000Z",
        "channelId": "UC9obdDRxQkmn_4YpcBMTYLw",
        "title": "CATS will make you LAUGH YOUR HEAD OFF - Funny CAT compilation",
        "description": "HELP HOMELESS ANIMALS AND WIN A PRIZE BY CHOOSING AN AWESOME ITEM FROM OUR STORE: https://fanjoy.co/collections/tiger-productions ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/hY7m5jjJ9mM/hqdefault.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "Tiger FunnyWorks",
        "liveBroadcastContent": "none"
      }
    }, {
      "kind": "youtube#searchResult",
      "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/u2n3wez7ljBkwPSV6WkGrkhsBlI\"",
      "id": {
        "kind": "youtube#video",
        "videoId": "5dsGWM5XGdg"
      },
      "snippet": {
        "publishedAt": "2016-12-24T10:30:00.000Z",
        "channelId": "UCKy3MG7_If9KlVuvw3rPMfw",
        "title": "Cats are so funny you will die laughing - Funny cat compilation",
        "description": "Cats are simply the funniest and most hilarious pets, they make us laugh all the time! Just look how all these cats & kittens play, fail, get along with dogs and ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/5dsGWM5XGdg/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/5dsGWM5XGdg/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/5dsGWM5XGdg/hqdefault.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "Tiger Productions",
        "liveBroadcastContent": "none"
      }
    }, {
      "kind": "youtube#searchResult",
      "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/0ej2DuHdmsgJLqZkGmegnoBrISU\"",
      "id": {
        "kind": "youtube#video",
        "videoId": "tntOCGkgt98"
      },
      "snippet": {
        "publishedAt": "2013-12-31T05:21:57.000Z",
        "channelId": "UCMW0kSYo44uRxvE21JqEWTQ",
        "title": "Funny Cats Compilation [Most See] Funny Cat Videos Ever Part 1",
        "description": "Funny Cats Compilation - Funny Cat Videos - Funny Videos Check out the NEW Part 2 Compilation!: https://youtu.be/XHr2cvkA8rQ Funny Cats Compilation Try ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/tntOCGkgt98/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/tntOCGkgt98/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/tntOCGkgt98/hqdefault.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "Forget Your Sadness",
        "liveBroadcastContent": "none"
      }
    }, {
      "kind": "youtube#searchResult",
      "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/z7OlabMtkYnYM6gpAKDiCkAEaEk\"",
      "id": {
        "kind": "youtube#video",
        "videoId": "njSyHmcEdkw"
      },
      "snippet": {
        "publishedAt": "2016-06-26T13:50:56.000Z",
        "channelId": "UCdmnOJ63TCe2AEdeFTYiweA",
        "title": "Funny Cats Compilation 2016  - Best Funny Cat Videos Ever || Funny Vines",
        "description": "CLICK NOW: https://goo.gl/TSNzGP for 25% discount T-Shirt in Sunfrog with code ILIKE25PLEEZ (Expired 31/10/17) 60$ OFF 200$ or more buyer total purchase ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/njSyHmcEdkw/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/njSyHmcEdkw/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/njSyHmcEdkw/hqdefault.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "Funny Vines",
        "liveBroadcastContent": "none"
      }
    }, {
      "kind": "youtube#searchResult",
      "etag": "\"cbz3lIQ2N25AfwNr-BdxUVxJ_QY/3BYpEDhVQCZ3B2FmPQVPbZK5FwI\"",
      "id": {
        "kind": "youtube#video",
        "videoId": "1Wh8RzcQZr4"
      },
      "snippet": {
        "publishedAt": "2017-05-02T12:00:00.000Z",
        "channelId": "UCPDis9pjXuqyI7RYLJ-TTSA",
        "title": "Hilarious Cat Fails (April 2017)",
        "description": "Enjoy these hilarious cat fails, some of them are just mean though!! Which feline is your favorite? Tell us in the comment section and don't forget to submit your ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/1Wh8RzcQZr4/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/1Wh8RzcQZr4/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/1Wh8RzcQZr4/hqdefault.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "FailArmy",
        "liveBroadcastContent": "none"
      }
    }
  ]
};

export const sampleData = {
  search: {
    picture: 'dog'
  },
  selectedVideo: {
    video: video,
    status: 'FULFILLED'
  },
  youtubeData: {
    myData: videos,
    status: 'FULFILLED'
  }
}
export const emptySampleData = {
  search: {
    picture: null,
  },
  selectedVideo: {
    video: null,
    status: null,
  },
  youtubeData: {
    myData: null,
    status: null,
  }
}
