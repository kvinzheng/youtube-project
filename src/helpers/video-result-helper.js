export const dataConvertor = array =>
  array.map(ele => ele.snippet);
