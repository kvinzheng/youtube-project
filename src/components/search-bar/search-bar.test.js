import React from 'react';
import {shallow, configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchBar from './search-bar';
import {mapStateToProps} from './search-bar-container';
import {mapDispatchToProps} from './search-bar-container';
import {sampleData} from '../../helpers/sample-data-test';

configure({adapter: new Adapter()});

const setSearchTerm = jest.fn();
const loadAllData = jest.fn();

describe('SearchBar Component', () => {
  it('should render without any data defined', () => {
    const component = shallow(<SearchBar loadAllData={loadAllData} setSearchTerm={setSearchTerm} searchTerm={''}/>);
    expect(component).toMatchSnapshot();
  });

  it('should render with props passed in', () => {
    const component = shallow(<SearchBar loadAllData={loadAllData} setSearchTerm={setSearchTerm} searchTerm={'cat'}/>);
    expect(component).toMatchSnapshot();
  });
  it('Renders with correct classes', () => {
    const component = shallow(<SearchBar loadAllData={loadAllData} setSearchTerm={setSearchTerm} searchTerm={'cat'}/>);
    expect(component.find('#search-bar').length).toBe(1);
    expect(component.find('.search-button').length).toBe(1);
  });

  it('mapStateToProps', () => {
    const state = sampleData;
    const expected = {
      searchTerm: sampleData.search.picture
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });
  it('mapStateToProps default data', () => {
    const state = {
      search: {
        picture: null
      },
      selectedVideo: {
        video: {},
        state: 'PENDING'
      },
      youtubeData: {
        myData: null,
        status: 'PENDING'
      }
    };
    const expected = {
      searchTerm: ''
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it('matDispatchtoProps', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('setSearchTerm');
    expect(mapDispatchToProps(dispatch)).toHaveProperty('loadAllData');
  });

  it('When submit on the form', () => {
    const component = mount(<SearchBar loadAllData={loadAllData} setSearchTerm={setSearchTerm} searchTerm={'cat'}/>);
    component.find('.search-form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(loadAllData).toHaveBeenCalled();
  });

  it('When type on the input', () => {
    const component = mount(<SearchBar loadAllData={loadAllData} setSearchTerm={setSearchTerm} searchTerm={'cat'}/>);
    component.find('#input-bar').simulate('change', {
      preventDefault: () => {},
      target: {
        value: 'cat'
      }
    });
    expect(loadAllData).toHaveBeenCalled();
  });

});
