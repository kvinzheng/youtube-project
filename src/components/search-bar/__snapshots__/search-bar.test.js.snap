// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component should render with props passed in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    loadAllData={[Function]}
    searchTerm="cat"
    setSearchTerm={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="search-form"
        onSubmit={[Function]}
>
        <input
                id="input-bar"
                onChange={[Function]}
                placeholder="Search your youtube videos here"
                type="text"
        />
        <button
                className="search-button"
        >
                Search
        </button>
</form>,
      "id": "search-bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="input-bar"
            onChange={[Function]}
            placeholder="Search your youtube videos here"
            type="text"
/>,
          <button
            className="search-button"
>
            Search
</button>,
        ],
        "className": "search-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "id": "input-bar",
            "onChange": [Function],
            "placeholder": "Search your youtube videos here",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "className": "search-button",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="search-form"
          onSubmit={[Function]}
>
          <input
                    id="input-bar"
                    onChange={[Function]}
                    placeholder="Search your youtube videos here"
                    type="text"
          />
          <button
                    className="search-button"
          >
                    Search
          </button>
</form>,
        "id": "search-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              id="input-bar"
              onChange={[Function]}
              placeholder="Search your youtube videos here"
              type="text"
/>,
            <button
              className="search-button"
>
              Search
</button>,
          ],
          "className": "search-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "id": "input-bar",
              "onChange": [Function],
              "placeholder": "Search your youtube videos here",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "className": "search-button",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SearchBar Component should render without any data defined 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    loadAllData={[Function]}
    searchTerm=""
    setSearchTerm={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="search-form"
        onSubmit={[Function]}
>
        <input
                id="input-bar"
                onChange={[Function]}
                placeholder="Search your youtube videos here"
                type="text"
        />
        <button
                className="search-button"
        >
                Search
        </button>
</form>,
      "id": "search-bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="input-bar"
            onChange={[Function]}
            placeholder="Search your youtube videos here"
            type="text"
/>,
          <button
            className="search-button"
>
            Search
</button>,
        ],
        "className": "search-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "id": "input-bar",
            "onChange": [Function],
            "placeholder": "Search your youtube videos here",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "className": "search-button",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="search-form"
          onSubmit={[Function]}
>
          <input
                    id="input-bar"
                    onChange={[Function]}
                    placeholder="Search your youtube videos here"
                    type="text"
          />
          <button
                    className="search-button"
          >
                    Search
          </button>
</form>,
        "id": "search-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              id="input-bar"
              onChange={[Function]}
              placeholder="Search your youtube videos here"
              type="text"
/>,
            <button
              className="search-button"
>
              Search
</button>,
          ],
          "className": "search-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "id": "input-bar",
              "onChange": [Function],
              "placeholder": "Search your youtube videos here",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "className": "search-button",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
